src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -162,21 +187,31 @@ function useChatUtilsHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatUtilsHook {
       channelData.channelPicture = channelImage;
       channelData.name = channelName;
 
-      await channelData.save(localDb);
-    }
-
-    const promise = originalMembers?.map((member) => {
-      return new Promise((resolve, reject) => {
-        const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
-        try {
-          user.saveOrUpdateIfExists(localDb).then(() => resolve(null));
-        } catch (e) {
-          reject(e);
+      queue.addPriorityJob({
+        label: `saveChannelData-${channel?.id}`,
+        priority: QueueJobPriority.MEDIUM,
+        task: async () => {
+          channelData.saveIfLatest(localDb);
         }
       });
-    });
+    }
+
+    originalMembers?.map((member) => {
+      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
+      try {
+        queue.addPriorityJob({
+          label: `save user from channel detail fetch${channel?.name}`,
+          priority: QueueJobPriority.MEDIUM,
+          task: async () => {
+            user.saveOrUpdateIfExists(localDb);
+          }
+        });
+      } catch (e) {
+        console.log('error on save user from channel detail fetch', e);
+      }
 
-    await Promise.all(promise).catch((e) => console.log('saveChannelDetail error', e));
+      return null;
+    });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The error handling in the `helperSaveChannelDetail` function could be improved. Currently, if an error occurs while saving user data from channel detail fetch, it is logged to the console but not handled further. This could lead to unexpected behavior in your application. Consider adding more robust error handling here.
```diff
    originalMembers?.map((member) => {
      const user = UserSchema.fromMemberWebsocketObject(member, channel?.id);
      try {
        queue.addPriorityJob({
          label: `save user from channel detail fetch${channel?.name}`,
          priority: QueueJobPriority.MEDIUM,
          task: async () => {
            user.saveOrUpdateIfExists(localDb);
          }
        });
      } catch (e) {
        console.log('error on save user from channel detail fetch', e);
        // Add additional error handling here
      }
 
      return null;
    });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->